Userland/Libraries/LibRegex/RegexParser.cpp

```diff
@@ -956,28 +956,37 @@ bool ECMA262Parser::parse_pattern(ByteCode& stack, size_t& match_length_minimum,
 
 bool ECMA262Parser::parse_disjunction(ByteCode& stack, size_t& match_length_minimum, bool unicode, bool named)
 {
-    ByteCode left_alternative_stack;
-    size_t left_alternative_min_length = 0;
-    auto alt_ok = parse_alternative(left_alternative_stack, left_alternative_min_length, unicode, named);
-    if (!alt_ok)
-        return false;
+    size_t total_match_length_minimum = NumericLimits<size_t>::max();
+    Vector<ByteCode> alternatives;
+    do {
```

Comment:
A little nit-picky, but:
```c++
do {
    // Some lines of code
} while (true);
```

To me, that makes it less immediately obvious that this loop will loop forever until we explicitly break from it. Whereras if I read:
```c++
while (true) {
    // Some lines of code
}
```

My brain immediately know what kind of loop I'm getting into